### ISSUE TITLE
[Feature Request] Provided Option for testConnectivity.py  to write Found GBT Phase Settings to a File

### ISSUE THREAD

**Issue by bdorney** [cms-gem-daq-project/vfatqc-python-scripts#225]
## Brief summary of issue

We should provide two sets of output information:

 1. The results of the phase scan (Step 4) should be written to a file if an option `--writePhases2File` is provided.
 2. The found good phases should be written if VFATs are all properly synchronized (Step 5) if an option `--writePhases2File` is provided.
  - Note if the user skips step 4 and starts only from step 5 then the phases should not be written as the required dictionary isn't populated.

### Types of issue

- [ ] Bug report (report an issue with the code)
- [x] Feature request (request for change which adds functionality)

### Writing Phase Scan Results to File During Step 4
If the option `--writePhases2File` is provided then you change:

https://github.com/cms-gem-daq-project/vfatqc-python-scripts/blob/d38b9746bad1ccbdd4056c2406c15dd0055def7c/testConnectivity.py#L276-L278

Such that `outputFile` is not `None`, see documentation:

https://github.com/cms-gem-daq-project/xhal/blob/ea12e8928c6a3496ca6454a820e986c140ae4a7a/python/reg_interface_gem/core/gbt_utils_extended.py#L42-L51

If the `--writePhase2File` is not provided then `outputFile` should remain `None`.

### Writing Found Phases If Step 5 Is Successful
If the option flag `--writePhases2File` is provided for each OH in `ohMask` the phase scan results should be written to a file.  This should be done only if step 5 is passed, so after line 416 but before the DAC scan is attempted:

https://github.com/cms-gem-daq-project/vfatqc-python-scripts/blob/d38b9746bad1ccbdd4056c2406c15dd0055def7c/testConnectivity.py#L415-L420

The good phase information is stored in the nested dictionary `dict_phases2Save`.

However if step 4 has been skipped (e.g. option `-f5` is provided) then the found good phases should not be written (since they will not be known), so after line 416 you should do something like:

```python
   if args.firstStep <= 4"
       for oh in range(nOHs):
            # skip masked OH's
            # write good phases in dict_phases2Save to file
```

## Current Behavior

The found GBT phases are not written to a file.

## Context (for feature requests)

We would need to know the good phases to store in the config DB for use with The Blaster (TM).

## Your Environment

* Version used: d38b9746bad1ccbdd4056c2406c15dd0055def7c

**Comment by bdorney** [comment]
@BenjaminRS this is needed by mid march.

**Comment by jsturdy** [comment]
Closed by #232
